site/specs/webrtc.yml

```diff
@@ -2421,6 +2520,201 @@ components:
+          onDisconnect: '#/components/schemas/onDisconnectEventBody'
+          onConnect: '#/components/schemas/onConnectEventBody'
+    baseEventBody:
+      type: object
+      description: the common properties shared by all events
+      required:
+        - event
+      properties:
+        event:
+          type: string
+          description: >-
+            the name of the event, and the condition that describes the body
+            that will be present
+          example: onSipDialogStateChange
+          enum:
+            - onSipDialogStateChange
+            - onLeave
+            - onDisconnect
+            - onConnect
+        timestamp:
+          type: string
+          description: the epoch UTC time, in milliseconds, when the event happened
+          example: '1628786234676'
+        participantId:
+          type: string
+          description: the ID of Participant on which the event was detected
+          example: be199214-1a03-402a-a7b6-d002470a0465
+        dialogId:
+          type: string
+          description: the ID of Participant on which the event was detected
+          example: be199214-1a03-402a-a7b6-d002470a0465
+        tag:
+          type: string
+          description: >-
+            the tag associated with the Participant on which the event was
+            detected.
+          example: hello-world-browser
+      discriminator:
+        propertyName: event
+    dialogEventData:
+      type: object
+      description: >
+        The dialogEventData element contains information about the event that
+        triggered the transition to the  current Dialog state. This data is
+        dependent on extrinsic events that impact the Dialog. An understanding
+        of the details of the Dialog Event may affect the business logic choices
+        for the receiving application.
+      properties:
+        eventReason:
+          type: string
+          description: A summary of why the event happened
+          example: FAR END BUSY
+        eventErrorReason:
+          type: string
+          description: If the event was an error, a text description of the error
+          example: Destination Server unavailable
+        eventErrorCode:
+          type: number
+          description: >-
+            An error number to help classifiy the error condition - retain this
+            for support
+          example: 1022
+    sipDialogEventBody:
+      allOf:
+        - $ref: '#/components/schemas/baseEventBody'
+        - $ref: '#/components/schemas/sipDialogEventContents'
+    onLeaveEventBody:
+      allOf:
+        - $ref: '#/components/schemas/baseEventBody'
+    onDisconnectEventBody:
+      allOf:
+        - $ref: '#/components/schemas/baseEventBody'
+    onConnectEventBody:
+      allOf:
+        - $ref: '#/components/schemas/baseEventBody'
+    sipDialogEventContents:
+      type: object
+      properties:
+        dialogStateData:
+          $ref: '#/components/schemas/sipDialogResponseStateData'
```

Comment:
Are you sure this is what you want in the dialogStateData in the onSipDialogStateChange event? The sipDialogResponseStateData has the to, from and userToUser. We are currently sending over the call ID, the fromSipUri, fromTag, toSipUri, toTag, recordRouteSipUri, and contactSipUri.

Do we need a new type with some assortment of values from above?

Reply:
Let's discuss

The intent was to mimic the basic behavior of othe endpoints where changes to the aggregate state including creation and change returned the entire externnaly visible state of the resource.